### PR TITLE
Migrate KMS to ASF

### PR DESCRIPTION
**Pull request by whummer**
* migrate KMS to ASF - minimally invasive lift&shift migration
* contains a likely fix for a key lookup issue in #5716

### CODE REVIEW

File: localstack/services/providers.py [hunk 1]
```diff
@@ -138,9 +138,22 @@ def kinesis():
 
 @aws_provider()
 def kms():
-    from localstack.services.kms import kms_listener, kms_starter
+    from localstack.services.kms import kms_starter
+    from localstack.services.kms.provider import KmsProvider
 
-    return Service("kms", listener=kms_listener.UPDATE_KMS, start=kms_starter.start_kms)
+    if kms_starter.KMS_PROVIDER == "kms-local":
+        return kms_local_kms()
```
Comment (whummer) [inline localstack/services/providers.py:145]: Actually, this does not work - needs fixing. (going to update the PR in a bit)

Comment (whummer) [reply localstack/services/providers.py]: Updated.

File: localstack/services/kms/provider.py [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+            raise InvalidGrantTokenException("Grant token OR (grant ID, key ID) must be specified")
+
+    def list_retirable_grants(
+        self,
+        context: RequestContext,
+        retiring_principal: PrincipalIdType,
+        limit: LimitType = None,
+        marker: MarkerType = None,
+    ) -> ListGrantsResponse:
+        region_details = KMSBackend.get()
```
Comment (alexrashed) [inline localstack/services/kms/provider.py:203]: As with `list_grants`, the previous implementation checked if the (required) parameter (in this case `retiring_principal`) was set (see comment above).

**Review by alexrashed — CHANGES_REQUESTED** [review]

File: localstack/services/kms/provider.py [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+    def describe_key(
+        self, context: RequestContext, key_id: KeyIdType, grant_tokens: GrantTokenList = None
+    ) -> DescribeKeyResponse:
+        try:
+            return call_moto(context)
+        except Exception:
+            key_pairs = KMSBackend.get().key_pairs
+            key = key_pairs.get(key_id)
+            if not key:
+                raise
```
[the remark this answers is not in the document]
Comment (whummer) [reply localstack/services/kms/provider.py]: Removed this obsolete patch, as discussed 👍

File: localstack/services/kms/provider.py [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+        grant[CREATION_DATE] = time.time()
+
+        region_details.grants[grant[GRANT_ID]] = grant
+        return CreateGrantResponse(GrantId=grant[GRANT_ID], GrantToken=grant[GRANT_TOKENS][0])
+
+    @handler("ListGrants", expand=False)
+    def list_grants(
+        self, context: RequestContext, list_grants_request: ListGrantsRequest
+    ) -> ListGrantsResponse:
+        region_details = KMSBackend.get()
```
[the remark this answers is not in the document]
Comment (whummer) [reply localstack/services/kms/provider.py:138]: Great catch. Definitely was mistakenly relying on the parser to do the check here! Updated.

**Review by alexrashed — APPROVED** [review]
Looks great! Awesome to see the next API being migrated to ASF! 🚀

**Pull request merged**